presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestAggregations.java

```diff
@@ -660,12 +660,71 @@ public void test15WayGroupBy()
     @Test
     public void testApproximateCountDistinct()
     {
+        // test date
+        assertQuery("SELECT approx_distinct(orderdate) FROM orders", "SELECT 2443");
+        assertQuery("SELECT approx_distinct(orderdate, 0.023) FROM orders", "SELECT 2443");
+
+        // test timestamp
+        assertQuery("SELECT approx_distinct(CAST(orderdate AS TIMESTAMP)) FROM orders", "SELECT 2384");
+        assertQuery("SELECT approx_distinct(CAST(orderdate AS TIMESTAMP), 0.023) FROM orders", "SELECT 2384");
+
+        // test timestamp with time zone
+        assertQuery("SELECT approx_distinct(CAST((orderdate) AS TIMESTAMP WITH TIME ZONE)) FROM orders", "SELECT 2384");
+        assertQuery("SELECT approx_distinct(CAST((orderdate) AS TIMESTAMP WITH TIME ZONE), 0.023) FROM orders", "SELECT 2384");
+
+        // test day to second interval
+        assertQuery("SELECT approx_distinct(\"$literal$interval day to second\"(custkey)) FROM orders", "SELECT 990");
```

Comment:
I was trying to find a way to create an interval dynamically (from a column value), but it seems that it is not possible. Also i don't think we ever store interval as a column. Frankly, i would just drop `appros_distinct` for that type for now.